crates/re_viewer/src/viewer_analytics.rs

```diff
@@ -90,29 +98,60 @@ impl ViewerAnalytics {
         self.record(Event::append("viewer_started".into()));
     }
 
-    pub fn on_new_recording(&mut self, msg: &re_log_types::BeginRecordingMsg) {
-        // We hash the application_id and recording_id unless this is an official example.
-        // That's because we want to be able to track which are the popular examples,
-        // but we don't want to collect actual application ids.
-        self.register("application_id", {
-            let prop = Property::from(msg.info.application_id.0.clone());
-            if msg.info.is_official_example {
-                prop
-            } else {
-                prop.hashed()
-            }
-        });
-        self.register("recording_id", {
-            let prop = Property::from(msg.info.recording_id.to_string());
-            if msg.info.is_official_example {
-                prop
-            } else {
-                prop.hashed()
+    /// When we have loaded the start of a new recording.
+    pub fn on_open_recording(&mut self, log_db: &re_data_store::LogDb) {
+        if let Some(rec_info) = log_db.recording_info() {
+            // We hash the application_id and recording_id unless this is an official example.
+            // That's because we want to be able to track which are the popular examples,
+            // but we don't want to collect actual application ids.
+            self.register("application_id", {
+                let prop = Property::from(rec_info.application_id.0.clone());
+                if rec_info.is_official_example {
+                    prop
+                } else {
+                    prop.hashed()
+                }
+            });
+            self.register("recording_id", {
+                let prop = Property::from(rec_info.recording_id.to_string());
+                if rec_info.is_official_example {
+                    prop
+                } else {
+                    prop.hashed()
+                }
+            });
+
+            let recording_source = match &rec_info.recording_source {
+                RecordingSource::Unknown => "unknown".to_owned(),
+                RecordingSource::PythonSdk(_version) => "python_sdk".to_owned(),
+                RecordingSource::RustSdk => "rust_sdk".to_owned(),
+                RecordingSource::Other(other) => other.clone(),
+            };
+
+            if let RecordingSource::PythonSdk(version) = &rec_info.recording_source {
+                // This will over-write what is set by `AppEnvironment`, if any.
+                // Usually these would be the same, but if you are using one python version to look
+                // at a recording from an older python version, then they will differ.
+                // For simplicity's sake we ignore this. We just want a rough idea of
+                // what python version our users use.
+                self.register("python_version", version.to_string());
```

Comment:
We can actually avoid this issue entirely: the python version that corresponds to the machine the user is running on (i.e. the machine that hosts the viewer) should never be included in events, it should only be part of the user profile (along with the rust version that's already there) i.e. `update_metadata`.

The `viewer_started` event then only keeps track of the app environment, while the `open_recording` event always keeps track of the `python_version` that corresponds to the machine doing the recording, never the host.
We can also put the rust version of the machine doing the recording in there, for consistency.

That way everything becomes consistent and relatively straightforward to reason about:
- The python & rust versions seen in user profiles (`update_metadata`) are those of the host machine (the one hosting the viewer).
- The python & rust versions seen in events correspond to the machine doing the recording.

I think this changes are trivial to add on top of everything you just did?
If we agree on that I'll just make another PR.

Reply:
sounds good!